Build errors during cmake with -DENABLE_CUFFT_BACKEND=True

Could someone help me with installing oneMKL with cufft?

Cmake --version: 3.16.3
Using open source dpcpp compiler: clang version 17.0.0

`**cmake .. -DCMAKE_CXX_COMPILER=/home/tcs/sycl_workspace/llvm/build/bin/clang++ -DCMAKE_C_COMPILER=/home/tcs/sycl_workspace/llvm/build/bin/clang -DENABLE_CUFFT_BACKEND=True -DENABLE_MKLCPU_BACKEND=False -DENABLE_MKLGPU_BACKEND=False**

-- CMAKE_BUILD_TYPE: Release
-- TARGET_DOMAINS: dft
-- Looking for dpc++
-- Looking for C++ include pthread.h
-- Looking for C++ include pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE
-- Found CUDA: /usr/local/cuda (found version "11.6")
-- Performing Test IS_SUPPORTED
-- Performing Test IS_SUPPORTED - Success
-- The C compiler identification is Clang 17.0.0
-- Check for working C compiler: /home/tcs/sycl_workspace/llvm/build/bin/clang
-- Check for working C compiler: /home/tcs/sycl_workspace/llvm/build/bin/clang -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Found PythonInterp: /opt/intel/oneapi/intelpython/latest/bin/python (found version "3.9.12")
-- SYCL_DEVICE_FILTER will be set to the following value(s): [gpu] for run-time dispatching examples
-- Configuring done
CMake Error at src/dft/backends/cufft/CMakeLists.txt:25 (add_library):
  Target "onemkl_dft_cufft" links to target "CUDA::cufft" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?


CMake Error at src/dft/backends/cufft/CMakeLists.txt:25 (add_library):
  Target "onemkl_dft_cufft" links to target "CUDA::cuda_driver" but the
  target was not found.  Perhaps a find_package() call is missing for an
  IMPORTED target, or an ALIAS target is missing?


CMake Error at src/dft/backends/cufft/CMakeLists.txt:25 (add_library):
  Target "onemkl_dft_cufft" links to target "CUDA::cufft" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?


CMake Error at src/dft/backends/cufft/CMakeLists.txt:25 (add_library):
  Target "onemkl_dft_cufft" links to target "CUDA::cuda_driver" but the
  target was not found.  Perhaps a find_package() call is missing for an
  IMPORTED target, or an ALIAS target is missing?


CMake Error at src/dft/backends/cufft/CMakeLists.txt:26 (add_library):
  Target "onemkl_dft_cufft_obj" links to target "CUDA::cufft" but the target
  was not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?


CMake Error at src/dft/backends/cufft/CMakeLists.txt:26 (add_library):
  Target "onemkl_dft_cufft_obj" links to target "CUDA::cuda_driver" but the
  target was not found.  Perhaps a find_package() call is missing for an
  IMPORTED target, or an ALIAS target is missing?


-- Generating done
CMake Generate step failed.  Build files cannot be regenerated correctly.`

Hello @FMarno,

Thanks for responding.
I found a machine where the cmake version is 3.22.1. Now I'm able to build it successfully but during the
 `cmake --install . --prefix <path_to_install_dir>` step I'm getting the following error. 
```
-- Installing: /home/tcs2/vidya/oneMKL/build/./lib/cmake/oneMKL/oneMKLTargets.cmake
-- Installing: /home/tcs2/vidya/oneMKL/build/./lib/cmake/oneMKL/oneMKLTargets-release.cmake
-- Installing: /home/tcs2/vidya/oneMKL/build/./lib/cmake/oneMKL/oneMKLConfig.cmake
-- Installing: /home/tcs2/vidya/oneMKL/build/./lib/cmake/oneMKL/oneMKLConfigVersion.cmake
CMake Error at cmake/cmake_install.cmake:46 (file):
  file INSTALL cannot find "/home/tcs2/vidya/oneMKL/cmake/FindMKL.cmake": No
  such file or directory.
Call Stack (most recent call first):
  cmake_install.cmake:77 (include)
```
could you please help me out in fixing this issue?

Thanks,
Vidya

--- a/src/dft/backends/cufft/CMakeLists.txt
+++ b/src/dft/backends/cufft/CMakeLists.txt
@@ -20,7 +20,6 @@
 set(LIB_NAME onemkl_dft_cufft)
 set(LIB_OBJ ${LIB_NAME}_obj)
 
-find_package(CUDAToolkit REQUIRED)
 
 add_library(${LIB_NAME})
 add_library(${LIB_OBJ} OBJECT
@@ -41,7 +40,14 @@ target_include_directories(${LIB_OBJ}
 
 target_compile_options(${LIB_OBJ} PRIVATE ${ONEMKL_BUILD_COPT} ${MKL_COPT})
 
-target_link_libraries(${LIB_OBJ} PRIVATE CUDA::cufft CUDA::cuda_driver)
+if (${CMAKE_VERSION} VERSION_LESS "3.17.0")
+  find_package(CUDA REQUIRED)
+  target_include_directories(${LIB_OBJ} PRIVATE ${CUDA_INCLUDE_DIRS})
+  target_link_libraries(${LIB_OBJ} PRIVATE cuda ${CUDA_CUFFT_LIBRARIES})
+else()
+  find_package(CUDAToolkit REQUIRED)
+  target_link_libraries(${LIB_OBJ} PRIVATE CUDA::cufft CUDA::cuda_driver)
+endif()
 
 target_link_libraries(${LIB_OBJ} PUBLIC ONEMKL::SYCL::SYCL ${MKL_LINK_SYCL})
 

